vpr/src/place/initial_noc_placement.cpp

```diff
@@ -179,11 +195,16 @@ static void noc_routers_anneal(const t_noc_opts& noc_opts) {
     update_noc_normalization_factors(costs);
     costs.cost = calculate_noc_cost(costs.noc_cost_terms, costs.noc_cost_norm_factors, noc_opts);
 
-    // Maximum distance in each direction that a router can travel in a move
-    // It is assumed that NoC routers are organized in a square grid.
-    // Each router can initially move within the entire grid with a single swap.
+    const auto& compressed_noc_grid = get_compressed_noc_grid();
+    const size_t n_noc_layers = compressed_noc_grid.get_layer_nums().size();
+
+    /* Maximum distance in each direction that a router can travel in a move
+     * It is assumed that NoC routers are organized in a square grid.
+     * Each router can initially move within the entire grid with a single swap.
+     * For 3D NoCs, it is assumed that each layer has the same number of routers,
```

Comment:
I'd clarify this: "The calculation below assumes that NoC routers are organized in a square grid; if it is a 3D architecture it also assumes each layer has the same number of routers. Breaking that assumption is OK, but the calculation may not compute the best initial range limit in that case."